### PR TITLE
Removed dependency on websocket++, minimized dependencies

### PR DESCRIPTION
**Pull request by AssemblyJohn**
## Describe your changes

Removed the strong dependency to websocket++ by adding the relevant headers to a 3rd_party folder, and making the library optional based on a cmake option. The hard-coded websocket++ constants were also removed.

## Issue ticket number and link

Linked core PR: https://github.com/EVerest/everest-core/pull/631

## Checklist before requesting a review
- [x] I have performed a self-review of my code
- [ ] I have made corresponding changes to the documentation
- [ ] If OCPP 2.0.1: I have updated the [OCPP 2.0.1 status document](https://github.com/EVerest/libocpp/tree/main/doc/ocpp_201_status.md)
- [ ] I read the [contribution documentation](https://github.com/EVerest/EVerest/blob/main/CONTRIBUTING.md) and made sure that my changes meet its requirements

### CODE REVIEW

File: lib/ocpp/common/websocket/websocket_libwebsockets.cpp
Comment (Pietfried) [inline lib/ocpp/common/websocket/websocket_libwebsockets.cpp:719]: Shouldnt this use the `code` argument?

File: include/ocpp/common/websocket/websocket.hpp
Comment (Pietfried) [inline include/ocpp/common/websocket/websocket.hpp:43]: Looks like the `std::error_code reason` is not used and can also be removed

**Review by Pietfried — CHANGES_REQUESTED** [review]
Looks good. Added a few remarks / suggestions.

File: lib/ocpp/common/websocket/websocket_libwebsockets.cpp
Comment (AssemblyJohn) [reply lib/ocpp/common/websocket/websocket_libwebsockets.cpp:719]: This works on a fire-and-forget mode. We just notice the thread that we want to close, and the thread will do the close as soon as possible when it sees fit. We are not waiting for the close in order to get a response. Is there any case in which we care how/when we close? Additional mutex/conditional variables will have to be added in order to block the calling thread until the close is done if required.

**Comment by AssemblyJohn** [comment]
Implemented comments. Some codacity issues are related to some 3rd _party code I wouldn't modify too much. This can't be merged with the required modifications in everest-core.

**Review by Pietfried — APPROVED** [review]
looks good from my side. Please wait for @hikinggrass approval as well before merging

**Comment by AssemblyJohn** [comment]
I'd also require a review on: https://github.com/EVerest/everest-core/pull/631

This can't be merged without that.

File: include/ocpp/common/types.hpp
Comment (hikinggrass) [inline include/ocpp/common/types.hpp:673]: Probably a good idea to link that this is coming from https://github.com/rstudio/websocket/blob/2161ee7de8f8782fae4d717c59e1292f5f5d9cf2/src/lib/websocketpp/close.hpp#L56

File: include/ocpp/common/websocket/websocket_uri.hpp
Comment (hikinggrass) [inline include/ocpp/common/websocket/websocket_uri.hpp:57]: Is that comment still relevant?

**Review by hikinggrass — CHANGES_REQUESTED** [review]

File: include/ocpp/common/websocket/websocket_uri.hpp
Comment (AssemblyJohn) [reply include/ocpp/common/websocket/websocket_uri.hpp]: I think in the future we should do it manually, without any 3rd party, so in that case it would still be relevant.

**Review by Pietfried — APPROVED** [review]
looks good to me except a few nits

**Review by hikinggrass — APPROVED** [review]

**Pull request merged**
